### PR TITLE
Remove E_STRICT and other PHP 5 related information

### PR DESCRIPTION
**Pull request by kamil-tekiela**

### CODE REVIEW

**Comment by kamil-tekiela** [comment]
The only thing I am unsure of is the new default for error_report. I left it as it was.

**Comment by Girgias** [comment]
As of PHP 8.0.0 the default is E_ALL

**Comment by kamil-tekiela** [comment]
@Girgias  Thanks, but we need to keep the changelog of that. I didn't know how to add the changelog as part of this PR.

File: reference/errorfunc/functions/error-reporting.xml [hunk 1]
```diff
@@ -119,19 +119,11 @@ ini_set('error_reporting', E_ALL);
...
-    <link linkend="ini.error-reporting">error_reporting</link> is enhanced
-    to include <constant>E_STRICT</constant> errors (and vice versa).
-   </simpara>
-  </warning>
   <tip>
    <simpara>
     Passing in the value <literal>-1</literal> will show every possible error,
     even when new levels and constants are added in future PHP versions. The
-    <constant>E_ALL</constant> constant also behaves this way as of PHP 5.4.
+    behaviour is equivalent to passing <constant>E_ALL</constant> constant.
```
Comment (cmb69) [inline reference/errorfunc/functions/error-reporting.xml:126]: Prefer American English spelling:
```suggestion
    behavior is equivalent to passing <constant>E_ALL</constant> constant.
```

**Comment by cmb69** [comment]
> I didn't know how to add the changelog as part of this PR.

If it's about the ini setting page, changelog entries can be added to the table at the top of the page.

File: language-snippets.ent [hunk 2]
```diff
@@ -1236,7 +1233,7 @@ is inserted with (e.g.) <function xmlns="http://docbook.org/ns/docbook">domnode_
 <!ENTITY dom.node.inserted 'This node will not show up in the document unless
 it is inserted with (e.g.) <function xmlns="http://docbook.org/ns/docbook">DOMNode::appendChild</function>.'>
 
-<!ENTITY dom.allowstatic '<para xmlns="http://docbook.org/ns/docbook">This method <emphasis>may</emphasis> be called statically, but will issue an <constant>E_STRICT</constant> error.</para>'>
+<!ENTITY dom.allowstatic '<para xmlns="http://docbook.org/ns/docbook">This method <emphasis>may</emphasis> be called statically, but will issue an <constant>E_DEPRECATED</constant> error.</para>'>
```
Comment (cmb69) [inline language-snippets.ent:1236]: It's E_DEPRECATED in PHP 7, but an `Error` as of PHP 8.0.0.

**Review by cmb69 — CHANGES_REQUESTED** [review]
Thank you! Except for `dom.allowstatic` this looks good to me.

File: language-snippets.ent (hunk 2 above)
Comment (kamil-tekiela) [reply language-snippets.ent]: Is this acceptable or should I mention that it triggers an error now?

Comment (cmb69) [reply language-snippets.ent]: I think it's better to state that it triggers an error now.

File: language-snippets.ent [hunk 3]
```diff
@@ -1236,7 +1233,9 @@ is inserted with (e.g.) <function xmlns="http://docbook.org/ns/docbook">domnode_
 <!ENTITY dom.node.inserted 'This node will not show up in the document unless
 it is inserted with (e.g.) <function xmlns="http://docbook.org/ns/docbook">DOMNode::appendChild</function>.'>
 
-<!ENTITY dom.allowstatic '<para xmlns="http://docbook.org/ns/docbook">This method <emphasis>may</emphasis> be called statically, but will issue an <constant>E_STRICT</constant> error.</para>'>
+<!ENTITY dom.allowstatic '<para xmlns="http://docbook.org/ns/docbook">Prior to PHP 8.0.0 this method
+<emphasis>could</emphasis> be called statically, but would issue an <constant>E_DEPRECATED</constant> error.
+As of PHP 8.0.0 calling this method statically emits an <constant>E_ERROR</constant> level error</para>'>
```
Comment (kamil-tekiela) [inline language-snippets.ent:1238]: How about now? @cmb69

Comment (cmb69) [reply language-snippets.ent:1238]: Hmm, isn't it rather an `Error` exception?

Comment (kamil-tekiela) [reply language-snippets.ent:1238]: It's not an exception. It's a an error of severity `E_ERROR`. Compare:
```
error_reporting(E_ALL & ~E_ERROR);
$doc = DOMDocument::loadXML('<root><node/></root>');
echo $doc->saveXML();
```
and 
```
try {
    $doc = DOMDocument::loadXML('<root><node/></root>');
    echo $doc->saveXML();
} catch (Exception $e) {
    echo $e->getMessage();
}
```
Of course, `Error` is a type of exception in PHP, but it does not inherit from `Exception`.

Comment (Girgias) [reply language-snippets.ent:1238]: But one can catch an `Error` whereas one cannot catch an E_ERROR

Comment (kamil-tekiela) [reply language-snippets.ent]: Changed to "throws an <classname>Error</classname> exception"

**Review by cmb69 — APPROVED** [review]
Thank you!

**Pull request merged**